sdk/tables/azure-data-tables/migration_guide.md

```diff
@@ -0,0 +1,298 @@
+table_client.delete_table()
+```
+
+#### List and Query
+
+In `azure-cosmosdb-table`:
+```python
+from azure.cosmosdb.table import TableService
+client = TableService(...)
+
+# In the azure-cosmosdb-table library, there is no query table method
+tables = list(service.list_tables())
+for table in tables:
+    print(table.name)
+```
+
+In `azure-data-tables`:
+```python
+from azure.data.tables import TableServiceClient
+
+service_client = TableServiceClient.from_connection_string(conn_str)
+
+for table in service_client.list_tables():
+    print(table.name)
+
+query_filter = "TableName eq @name1 or TableName gt @name2'
+parameters = {
+    "name1": "myTableName1",
+    "name2": "myTableName2",
+}
+
+for table in service_client.query_tables(query_filter, parameters=parameters):
+    print(table.name)
+```
+
+### Entity Level Scenarios
+
+#### Insert and Delete
+
+In `azure-cosmosdb-table`:
+```python
+from azure.cosmosdb.table import TableService
+client = TableService(...)
+table_name = "tableName"
+entity = {
+    "PartitionKey": "pk0001",
+    "RowKey": "rk0001",
+    "StringProperty": "stringystring",
+    "BooleanProperty": False,
+    "IntegerProperty": 31,
+    "FloatProperty": 3.14159,
+    "BinaryProperty": b"binary",
+    "GuidProperty": uuid.uuid4(),
+    "DatetimeProperty": datetime.datetime.now(),
+}
+
+etag = client.insert_entity(table_name, entity)
+```
+
+In `azure-data-tables`:
+```python
+import datetime
+import os
+import uuid
+from azure.data.tables import TableServiceClient
+from azure.identity import AzureNamedKeyCredential
+
+entity = {
+    "PartitionKey": "pk0001",
+    "RowKey": "rk0001",
+    "StringProperty": "stringystring",
+    "BooleanProperty": False,
+    "IntegerProperty": 31,
+    "FloatProperty": 3.14159,
+    "BinaryProperty": b"binary",
+    "GuidProperty": uuid.uuid4(),
+    "DatetimeProperty": datetime.datetime.now(),
+}
+
+endpoint = "https://{}.table.cosmos.azure.com".format(account_name)
+table_name = "myTable"
+credential = AzureNamedKeyCredential(
+    key=os.environ["tables_primary_cosmos_account_key"],
+    name=os.environ["tables_cosmos_account_name"]
+)
+
+table_client = TableClient(endpoint, table_name, credential)
+resp = table_client.create_entity(entity)
+
+# To delete an entity, you can provide either an entire entity or a partition key and a row key
+table_client.delete_entity(entity)
+# OR
+pk = "pk_to_delete"
+rk = "rk_to_delete"
+table_client.delete_entity(pk, rk)
+```
+
+#### Update and Upsert
+
+In `azure-cosmosdb-table`:
+```python
+from azure.cosmosdb.table import TableService
+client = TableService(...)
+
+my_entity["Value"] += 5
+my_table = "tableName"
+
+etag = client.update_entity(my_table, my_entity)
+```
+
+In `azure-data-tables`:
+```python
+from azure.data.tables import UpdateMode
+
+my_entity["Value"] += 5
+table_client.update_entity(my_entity)
+
+my_entity["StringProperty"] = "new_string"
+table_client.upsert_entity(my_entity, mode=UpdateMode.MERGE)
```

Comment:
I think we should have comparitive lines for each of the update scenarios - how to do merge vs replace has changed in T1 vs T2 and that is not illustrated here.